### DISCUSSION TITLE
Visit runCatching stetement

### DISCUSSION THREAD

**Discussion by mzglinicki** [detekt/detekt#3923] [Q&A] [answered]
I would like to add a custom rule to check all catch statements in my code. I can check `try {} catch {}` using  the `override fun visitCatchSection(catchClause: KtCatchClause)` method. 

Is there any way to visit a runCatching statement?

```
class Example {
    suspend fun funWithGetOrElse() {
        runCatching { /* some code */ }
            .getOrElse {
                /* I would like to visit this statement */
            }
    }
}
```

**Comment by cortinico** [comment, marked as answer]
> Is there any way to visit a runCatching statement?

This function should do the job for you:

```kotlin
override fun visitLambdaExpression(lambdaExpression: KtLambdaExpression)
```

You should be able to filter for `runCatching` only by accessing `lambdaExpression.functionLiteral`

**Reply by mzglinicki** [reply]
@cortinico Thanks, works for me.
